### ISSUE TITLE
Stuck on loading repository

### ISSUE THREAD

**Issue by reaccessort** [gitthermal/thermal#3575]
I am always getting stuck screen for any repository
These all reporsitories have lots of commits because I have worked over 1 year
![image](https://user-images.githubusercontent.com/55362510/66451253-2fd11d00-ea8e-11e9-827f-caeaaad585ae.png)

**Comment by mittalyashu** [comment]
We're aware of that issue and working on it.

Reference #170

**Comment by Jangmingjin** [comment]
Cool

**Comment by neemanjabu** [comment]
Hi, what is the status of this issue, since it is a blocking issue?

**Comment by mittalyashu** [comment]
I am still working on it, you can subscribe to this issue for further updates :)

**Comment by smeijer** [comment]
Same issue, it doesn't work on Ubuntu 18.10. I do see commits being made in this repo. Does that mean that Thermal does work on other systems?

**Comment by nblackburn** [comment]
Same issue for me

**Comment by Abraxelx** [comment]
I'm usşng Linux and i have same trouble

**Comment by mittalyashu** [comment]
@nblackburn The  error message you have shared above it surely an error thrown by the application but it is not something related to the loading repository.

It throws an error when there is no remote URL to access from.

I understand that's not really best way to shown an error. I will look into, how error logging can be improved further.

**Comment by mittalyashu** [comment]
> Same issue, it doesn't work on Ubuntu 18.10. I do see commits being made in this repo. Does that mean that Thermal does work on other systems?

@smeijer The app itself work across platform, the error you are facing is more of an git related error _(git related error doesn't means there is something wrong with your machine or git which is installed)_, this issue is within Thermal app which uses Git APIs.

**Comment by nblackburn** [comment]
@mittalyashu I found the error by enabling the dev tools so this isn't something that was exposed to me. While this might not be related directly it  was all I have to hand on the issue.

**Comment by R4M5E5** [comment]
Are there any updates on this problem? Im facing the same problem.

**Comment by KaKi87** [comment]
Honestly, use SmartGit or Fork. @R4M5E5

**Comment by R4M5E5** [comment]
> Honestly, use SmartGit or Fork. @R4M5E5

Why? I wan't to use this client because it looks good.

**Comment by KaKi87** [comment]
Yes it does, and I was in the same mood as you when I arrived here.
But, it turns out the product is completely unfinished.
I contributed, but things doesn't go fast enough.
So I gave up.
There is still a year of full time development needed to finish the first stable version of Thermal.

**Comment by mittalyashu** [comment]
@reaccessort @Jangmingjin @neemanjabu @smeijer @nblackburn @Abraxelx @R4M5E5 @KaKi87 

Please read this issue #3614
